### PR TITLE
Upgrading formatters and pre-commit hooks

### PR DESCRIPTION
**Pull request by Alexandra-Green**
## What this PR accomplishes

- Upgrades linter to version and fixes formatting issues of previous versions
- Add pre-commit hook to lint code
- Adds settings.json.default file with vscode css formatter settings and helpful editing settings

## Issue(s) addressed

- djLint trans tag indentation issue -> trans tags are now blocks.
- Other djLint formatter bug fixes based on their 1.23.0 release
- Added pre-commit hook to use djLint to lint code.
- Added CSS formattter

## What needs review

- pre-commit hook -> `pip install -r dev-requirements.txt`
   - run command `pre-commit install`
   - Remove the first line of `ckanext/ontario_theme/templates/internal/home/snippets/ontario_theme_intro.html` (the djlint comment)
   - stage changes and attempt to commit the change
   - Pre-commit should prevent the file to be committed
   - Undo changes to `ckanext/ontario_theme/templates/internal/home/snippets/ontario_theme_intro.html` and add a comment into file
   - save changes and attempt to commit 
   - Pre-commit should run and be successful

- CSS formatter
   - Create file `settings.json` in the `.vscode` folder
   - Copy contents of `settings.json.default` into the `settings.json` file. Save.
   - In `commom.css` copy this and save file. There should be 2 errors, and 1 warning in the bottom right of your vscode, click on it to view the errors
``` 
.module h1>p, .module h1{
  margin: 0px;
  color: #0000000;
} 
```
*error message:*
errors

   - The file should format to this
``` 
.module h1 > p,
.module h1 {
  margin: 0px;
  color: #0000000;
}
```
   - remove the "px" from margin and remove a zero from color. The errors should be gone. **Remove the code from the file**

**Notes:**
- I only left generic settings in the `settings.json.default`

### CODE REVIEW

File: ckanext/ontario_theme/i18n/fr/LC_MESSAGES/ckanext-ontario_theme.mo
Comment (aimalkhan) [inline ckanext/ontario_theme/i18n/fr/LC_MESSAGES/ckanext-ontario_theme.mo:1]: is that an old file we are removing?

File: .gitignore [hunk 1]
```diff
@@ -45,4 +45,5 @@ docs/_build/
 gtm.html
 gtm_ns.html
 
-.vscode/
+.vscode/**/*
```
Comment (aimalkhan) [inline .gitignore:48]: Is ignoring everything under `.vscode` with an exception of `settings.json.default` needed? Are there any other files inside inside the `.vscode` directory?

**Review by aimalkhan — COMMENTED** [review]
- do we want to include djhtml in the commit hooks?
- will adding djlint to commit hooks run it under the env.? we wanted to run it outside of the env for ckan2.9.x

i am still testing out the change in my local env, and will +1 once done

File: ckanext/ontario_theme/i18n/fr/LC_MESSAGES/ckanext-ontario_theme.mo
Comment (Alexandra-Green) [reply ckanext/ontario_theme/i18n/fr/LC_MESSAGES/ckanext-ontario_theme.mo:1]: No. It's unclear why it says -163 Bytes (100%), but the .mo file still has 230 translation files. Can you run `python3 setup.py compile_catalog` and see if you get the output `230 of 232 messages (99%) translated in ckanext/ontario_theme/i18n/fr/LC_MESSAGES/ckanext-ontario_theme.po`

File: .gitignore (hunk 1 above)
Comment (Alexandra-Green) [reply .gitignore:48]: Let me know whether you think this is needed after the tests on your local.

**Comment by Alexandra-Green** [comment]
>     * will adding djlint to commit hooks run it under the env.? we wanted to run it outside of the env for ckan2.9.x

No, pre-commit hooks initializes an environment for each hook the first time it is run. [Pre-commit hooks docs](https://pre-commit.com/#pre-commit-install-hooks), this section has a little more info on it. But we can either run it outside or inside the env, it won't affect ckan2.9.x (from what I have tested, my env is still PyYAML 5.4.1)

**Comment by Alexandra-Green** [comment]
> * do we want to include djhtml in the commit hooks?

With the new updates to djlint, we won't need djhtml much longer, There are still some areas where djlint lacks with formatting that djhtml makes up for. Currently only the linter is in the commit hooks and formatting is done by the dev.

File: .gitignore (hunk 1 above)
Comment (aimalkhan) [reply .gitignore:48]: follows recommended pattern.

File: ckanext/ontario_theme/i18n/fr/LC_MESSAGES/ckanext-ontario_theme.mo
Comment (aimalkhan) [reply ckanext/ontario_theme/i18n/fr/LC_MESSAGES/ckanext-ontario_theme.mo:1]: image

**Review by aimalkhan — APPROVED** [review]
documentation to be added in a future PR.
good job with this change 👍

**Pull request merged**